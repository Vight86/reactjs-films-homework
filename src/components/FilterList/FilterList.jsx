import React from 'react';
import propTypes from 'prop-types';
import styles from './FilterList.scss';

const FilterList = ({ children }) => (
  <nav className={styles.filterList}>
    {children.map((item) => item)}
  </nav>
);

FilterList.propTypes = {
  children: propTypes.arrayOf(propTypes.element),
};

FilterList.defaultProps = {
  children: [
    <button type="button" key="key1">Button</button>,
    <button type="button" key="key2">Button</button>,
  ],
};
export default FilterList;
