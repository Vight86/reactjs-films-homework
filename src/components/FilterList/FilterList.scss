.filterList {
  margin-bottom: -1px;

  & > ul {
    padding: 0;
    margin: 0;
    list-style: none;

    & > li {
      display: inline-block;
    }
  }
}
