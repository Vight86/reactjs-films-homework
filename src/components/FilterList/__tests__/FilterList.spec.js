import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FilterList from '../index';

const setUp = (children) => {
  const renderer = new ShallowRenderer();
  renderer.render(<FilterList>{children}</FilterList>);
  return renderer.getRenderOutput();
};

describe('render FilterList component', () => {
  it('render without props', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });

  it('render with props', () => {
    const result = setUp([
      <button type="button" key="key1">Button 1</button>,
      <button type="button" key="key2">Button 2</button>,
      <button type="button" key="key3">Button 3</button>,
      <button type="button" key="key4">Button 4</button>,
    ]);
    expect(result).toMatchSnapshot();
  });
});
