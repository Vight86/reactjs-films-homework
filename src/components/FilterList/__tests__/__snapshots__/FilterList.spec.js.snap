// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FilterList component render with props 1`] = `
<nav
  className="filterList"
>
  <button
    type="button"
  >
    Button 1
  </button>
  <button
    type="button"
  >
    Button 2
  </button>
  <button
    type="button"
  >
    Button 3
  </button>
  <button
    type="button"
  >
    Button 4
  </button>
</nav>
`;

exports[`render FilterList component render without props 1`] = `
<nav
  className="filterList"
>
  <button
    type="button"
  >
    Button
  </button>
  <button
    type="button"
  >
    Button
  </button>
</nav>
`;
